docs/designs/xref.md

```diff
@@ -0,0 +1,103 @@
+# Xref
+- User can define uid with some properties in markdown, JSON and YAML files. Then output xrefmap.json for others to reference to. 
+- User can reference to uids in markdown, JSON and YAML files, they will be resolved during `docfx build`.
+
+## Define uid
+- User can define uid in YAML header of markdown file
+```
+---
+uid: a
+title: ASP.NET Documentation
+description: Learn how to develop ASP.NET and ASP.NET web applications. Get documentation, example code, tutorials, and more.
+---
+```
+- User can also define uid in JSON
+```json
+{
+  "$schema": "https://raw.githubusercontent.com/dotnet/docfx/v3/schemas/TestData.json",
+  "uid": "a",
+  "title": "ASP.NET Documentation",
+  "description": "Learn how to develop ASP.NET and ASP.NET web applications. Get documentation, example code, tutorials, and more."
+}
+```
+- And define uid in YAML
+```yml
+#YamlMime:TestData
+uid: a
+title: ASP.NET Documentation
+description: Learn how to develop ASP.NET and ASP.NET web applications. Get documentation, example code, tutorials, and more.
+```
+
+## Output xref map
```

Comment:
Maybe add some details or open questions here: For a MarkdownAttribute (or other attribute need trasform), whether the value in output xref map should be before or after transform?